web/profiles/joinup/joinup.behat.inc

```diff
@@ -0,0 +1,163 @@
+    $values = [];
+    $aliases = self::userFieldAliases();
+    // Replace the column aliases with the actual field names.
+    foreach ($user_table->getRowsHash() as $key => $value) {
+      if (array_key_exists($key, $aliases)) {
+        $values[$aliases[$key]] = $value;
+      }
+      else {
+        throw new \Exception("Unknown column '$key' in user table.");
+      }
+    };
+    $this->createUser($values);
+  }
+
+  /**
+   * Mapping of human readable names to machine names.
+   *
+   * @return array
+   *    Field mapping.
+   */
+  private static function userFieldAliases() {
+    return [
+      'Username' => 'name',
+      'Password' => 'pass',
+      'E-mail' => 'mail',
+      'Status' => 'status',
+      'First name' => 'field_user_first_name',
+      'Family name' => 'field_user_family_name',
+      'Photo' => 'field_user_photo',
+      'Professional profile' => 'field_user_professional_profile',
+      'Organisation' => 'field_user_organisation',
+      'Nationality' => 'field_user_nationality',
+      'Professional domain' => 'field_user_professional_domain',
+      // @todo Social network
+    ];
+  }
+
+  /**
+   * Creates a user from the given property and field data.
+   *
+   * @param array $values
+   *   An optional associative array of values, keyed by property name.
+   *
+   * @return \Drupal\user\Entity\User
+   *   A new user entity.
+   *
+   * @throws \Exception
+   *   Thrown when a given image is not found.
+   */
+  public function createUser(array $values) {
+    // Add image.
+    if (!empty($values['field_user_photo'])) {
+      $filename = $values['field_user_photo'];
+      $files_path = $this->getMinkParameter('files_path');
+      $path = rtrim(realpath($files_path), DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR . $filename;
+      if (!is_file($path)) {
+        throw new \Exception("Image '$filename' was not found in file path '$files_path'.");
+      }
+      // Copy the image into the public files folder and turn it into a File
+      // entity before linking it to the collection.
+      $uri = 'public://' . $filename;
+      file_unmanaged_copy($path, $uri);
+      $image = File::create(['uri' => $uri]);
+      $image->save();
+      $values['field_user_photo'] = [$image->id()];
+
+      $this->images[$image->id()] = $image;
+    }
+    $user = User::create($values);
+    $user->save();
+    $this->users[$user->id()] = $user;
+
+    return $user;
+  }
+
+  /**
+   * Checks the users existence.
+   *
+   * @param string $username
+   *   The username of the user.
+   *
+   * @throws \Exception
+   *   Thrown when the user is not found.
+   *
+   * @Then I should have a :username user
+   */
+  public function assertUserExistence($username) {
+    $user = user_load_by_name($username);
+
+    if (empty($user)) {
+      throw new \Exception("Unable to load expected user " . $username);
+    }
+  }
+
+  /**
+   * Remove any created users.
+   *
+   * @AfterScenario
+   */
+  public function cleanUsers() {
```

Comment:
There is already a cleanUsers inherited here but it expectes stdObjects rather than User objects.